### PR TITLE
Rework drop aggro

### PR DESCRIPTION
**Pull request by Pbeitelmann**

### CODE REVIEW

File: src/game/ThreatManager.cpp [hunk 1]
```diff
@@ -286,67 +290,61 @@ void ThreatContainer::update()
...
-		2637, 18657, 18658,
-		//Psychic Scream
-		8122, 8124, 10888, 10890, 13704, 15398, 22884, 26042, 27610, 34322, 43432,
-		//Shackle
-		9484, 9485, 10955, 11444, 38051, 38505, 40135
+// Check the unit for mechanics that should cause an aggro drop
+bool CheckForAggroDropMechanics(Unit* target) {
+	bool aggroDrop = false;
+	uint32 mechanicArray[] = { 
+		1,	//charmed checked
```
Comment (robinsch) [inline src/game/ThreatManager.cpp]: magic numbers

File: src/game/ThreatManager.cpp [hunk 2]
```diff
@@ -286,67 +290,61 @@ void ThreatContainer::update()
...
-		//Hibernate
-		2637, 18657, 18658,
-		//Psychic Scream
-		8122, 8124, 10888, 10890, 13704, 15398, 22884, 26042, 27610, 34322, 43432,
-		//Shackle
-		9484, 9485, 10955, 11444, 38051, 38505, 40135
+// Check the unit for mechanics that should cause an aggro drop
+bool CheckForAggroDropMechanics(Unit* target) {
+	bool aggroDrop = false;
+	uint32 mechanicArray[] = {
```
Comment (robinsch) [inline src/game/ThreatManager.cpp]: enum mechanic instead of uint32

File: src/game/ThreatManager.cpp [hunk 3]
```diff
@@ -286,67 +290,61 @@ void ThreatContainer::update()
...
+	//Some spells are excluded from this check
+	uint32 spellExceptions[] = {
+		29511	//Maiden of Virtue Repentance
+	};
+
+	uint32 size = (sizeof(spellExceptions) / sizeof(*spellExceptions));
 	for (int i = 0; i < (size); ++i) {
-		if (target->HasAura(spellArray[i]) == true) {
-			return true;
+		if (target->HasAura(spellExceptions[i]) == true) {
```
Comment (robinsch) [inline src/game/ThreatManager.cpp]: == true

File: src/game/ThreatManager.cpp [hunk 4]
```diff
@@ -286,67 +290,61 @@ void ThreatContainer::update()
...
+	//Get Auras and look for qualifying mechanics	
+	Unit::AuraMap& Auras = target->GetAuras();
+		for (Unit::AuraMap::iterator iter = Auras.begin(); iter != Auras.end() && aggroDrop == false; ++iter)
+		{
+			SpellEntry const* spell = iter->second->GetSpellProto();
+			if (spell->Mechanic != 0)
+			{
+				aggroDrop = std::find(std::begin(mechanicArray), std::end(mechanicArray), spell->Mechanic) != std::end(mechanicArray);
+			}
+			if (spell->EffectMechanic[0] != 0)
```
Comment (robinsch) [inline src/game/ThreatManager.cpp]: loop

File: src/game/ThreatManager.cpp [hunk 5]
```diff
@@ -394,7 +392,7 @@ bool DropAggro(Creature* pAttacker, Unit * target)
         return true;
 
 	//Check for spells that should cause an aggro drop
-	if (CheckForAggroDropSpells(target) == true)
+	if (CheckForAggroDropMechanics(target) == true)
```
Comment (robinsch) [inline src/game/ThreatManager.cpp]: == true

**Comment by robinsch** [comment]
code style

File: src/game/ThreatManager.cpp (hunk 2 above)
Comment (Pbeitelmann) [inline src/game/ThreatManager.cpp]: As I said, I have no interest in working with you on any level, nor am I interested in your opinion. Feel free to provide your invaluable feedback somewhere else. Thanks.

**Comment by Pbeitelmann** [comment]
As I said, I have no interest in working with you on any level, nor am I interested in your opinion. Feel free to provide your invaluable feedback somewhere else. Thanks.

**Pull request merged**